block.go

```diff
@@ -259,7 +260,14 @@ func writeMetaFile(dir string, meta *BlockMeta) error {
 	if err := f.Close(); err != nil {
 		return err
 	}
-	return renameFile(tmp, path)
+
+	if err := fileutil.Rename(tmp, path); err != nil {
```

Comment:
@krasi-georgiev should replace `renameFile` with `fileutil.Rename`?

Maybe we should check all callers of `renameFile`:
1. repalce `renameFile` with `fileutil.Rename` for renaming a file.
2. repalce `renameFile` with `fileutil.Replace` for renaming a dir.

Reply:
Ahh good catch. I think option 1 looks best here.
Replace all instances of `renameFile` with `filetuil.Rename`, 
remove `renameFile`,
check that all callers of `filetuil.Rename` handle an error properly by deleting any left over files/dirs.



not sure why `renameFile` has this 
```
if err := os.RemoveAll(to); err != nil {
		return err
	}
```
`os.Rename` should handle this so it shouldn't be needed. Maybe double check to confirm.

Reply:
updated.

Reply:
I think best to put it in a defer statement similar to the one in `func (c *LeveledCompactor) write`

Reply:
ok